packages/toolpad-app/src/runtime/ToolpadApp.tsx

```diff
@@ -872,6 +878,7 @@ const queryClient = new QueryClient({
   defaultOptions: {
     queries: {
       retry: false,
+      staleTime: 10 * 60 * 1000,
```

Comment:
I think that we would be OK with:

```suggestion
      staleTime: 1000,
```

From what I understand, it would mean that we never query more than once every second, so 3600 requests per hour, below the 5000 limits of GitHub. It's not pretty, but I wouldn't expect anyone to want to refresh more than once per second. But for example in https://master--toolpad.mui.com/_toolpad/app/cl6rqzry10009arlv9sto6qja/pages/ip23ggo I would definitely want to use the refresh button after I label a few issues.

Reply:
> it would mean that we never query more than once every second

But it's not an unrealistic scenario to have more than one github queries on one page. Perhaps we can make it slightly longer, say 10s? I'm also planning to make this configurable at some point (https://github.com/mui/mui-toolpad/issues/1093).

Reply:
From my perspective, when using a refresh button, we should feel like the results returned are up to date, **reliably**. I doubt that a value > 1s can allow this UX. A second concern is that if we set 10s, then we might get blind to the problem. What I think would be great is to first solve the over-fetching, and then consider caching, but to solve slow queries, not overfetching.

Reply:
If I understand correctly, when you call `refetch` it should ignore cache and always refetch. So the button should always work, regardless of how long we make the `staleTime`.

Reply:
Oh yeah, this would be even better

Reply:
It should be already working this way though I'm not sure I explicitly tried this.

Reply:
@oliviertassinari I took care of the cache busting in the editor. Do you still think we should shorten the `staleTime` when we have the `refetch` function?

Reply:
@Janpot Per https://github.com/mui/mui-toolpad/issues/1163#issuecomment-1285114257, I think that 1000 is enough, sorry, I might have put you on the wrong track.

Reply:
> e.g. when editing records

The current architecture doesn't help with that neither. You'd still have to call `refetch` after updating the row to make the table up to date. There is no way to do it automatic as the list/delete/update methods are not linked in any way. For that we'd need something like a collection that we can link as a whole to a datagrid (https://github.com/mui/mui-toolpad/issues/385)

> But to be honest, I don't fully understand what this property does so I might not be the best one to answer.

`staleTime` sets the time after which data is considered out of date. That doesn't mean `react-query` will automatically refetch it after this time. It just means that after this time has passed, when certain triggers fire (e.g. window focus) `react-query` will refetch the query in the background.

Reply:
@Janpot Ok, then yes `10s` sounds great. It's large enough to have an impact and small enough to not completely go wrong.

Reply:
Adding feature request that could help with invalidation: https://github.com/mui/mui-toolpad/issues/1184